### PR TITLE
bump resource limits for Envoy Gateway deployment

### PR DESCRIPTION
**Pull request by arkodg**
Relates to https://github.com/envoyproxy/gateway/issues/1613

Must be revisited again when https://github.com/envoyproxy/gateway/issues/1365 is completed

### CODE REVIEW

**Review by zirain — APPROVED** [review]

**Comment by qicz** [comment]
ref https://github.com/envoyproxy/gateway/issues/1613#issuecomment-1619288673

File: charts/gateway-helm/values.tmpl.yaml [hunk 1]
```diff
@@ -7,10 +7,10 @@ deployment:
     resources:
       limits:
         cpu: 500m
-        memory: 128Mi
+        memory: 1024Mi
       requests:
-        cpu: 10m
-        memory: 64Mi
+        cpu: 100m
```
Comment (qicz) [inline charts/gateway-helm/values.tmpl.yaml:12]: the `requests` does not need to increase. EG can run on cpu 10m and mem 64Mi. for the limits IMO can't fix memory hign usage. https://github.com/envoyproxy/gateway/issues/1613#issuecomment-1619288673

Comment (qicz) [reply charts/gateway-helm/values.tmpl.yaml:12]: I see, but request changing is unnecessary for my production env.

Comment (arkodg) [reply charts/gateway-helm/values.tmpl.yaml:12]: I feel like `10m` is too low to start off with

**Review by Xunzhuo — APPROVED** [review]

**Pull request merged**